From 42d2d4faee38d2ed86fd8bc400e934dabd020de8 Mon Sep 17 00:00:00 2001
From: smitsohu <smitsohu@gmail.com>
Date: Mon, 18 Jul 2022 01:00:08 +0200
Subject: [PATCH] chroot hotfix

---
 src/firejail/fs_etc.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/firejail/fs_etc.c b/src/firejail/fs_etc.c
index 77fa00d6b..e82cb01ed 100644
--- a/src/firejail/fs_etc.c
+++ b/src/firejail/fs_etc.c
@@ -267,6 +267,9 @@ void fs_rebuild_etc(void) {
 	if (cfg.dns1 == NULL && !any_dhcp())
 		have_dhcp = 0;
 
+	if (have_dhcp == 0 && cfg.chrootdir)
+		return;
+
 	if (arg_debug)
 		printf("rebuilding /etc directory\n");
 	if (mkdir(RUN_DNS_ETC, 0755))
-- 
2.35.3

