From fd495aca14d17cc3ebd3f907bb7de2acc23cfeb7 Mon Sep 17 00:00:00 2001
From: Finn Rayment <finn@rayment.fr>
Date: Fri, 15 Jul 2022 13:57:54 +1000
Subject: [PATCH 1/1] 03

---
 config.def.h | 7 +++++++
 slock.c      | 6 ++++++
 2 files changed, 13 insertions(+)

diff --git a/config.def.h b/config.def.h
index 7877a1b..c2c1f4c 100644
--- a/config.def.h
+++ b/config.def.h
@@ -15,6 +15,13 @@ static const int failonclear = 1;
 /* time in seconds before the monitor shuts down */
 static const int monitortime = 5;
 
+/* number of failed password attempts until failcommand is executed.
+   Set to 0 to disable */
+static const int failcount = 0;
+
+/* command to be executed after [failcount] failed password attempts */
+static const char *failcommand = "shutdown -h now";
+
 static const int logosize = 75;
 static const int logow = 12; /* Grid width and height for right center alignment*/
 static const int logoh = 6;
diff --git a/slock.c b/slock.c
index 8e2c5b6..7ae2cb8 100644
--- a/slock.c
+++ b/slock.c
@@ -29,6 +29,7 @@
 #include "util.h"
 
 char *argv0;
+int failtrack = 0;
 
 enum {
     BACKGROUND,
@@ -205,6 +206,11 @@ readpw(Display *dpy, struct xrandr *rr, struct lock **locks, int nscreens,
 				if (running) {
 					XBell(dpy, 100);
 					failure = 1;
+					failtrack++;
+
+					if (failtrack >= failcount && failcount != 0){
+						system(failcommand);
+					}
 				}
 				explicit_bzero(&passwd, sizeof(passwd));
 				len = 0;
-- 
2.35.1

